packages/wrangler/src/secret/index.ts

```diff
@@ -444,9 +444,11 @@ export const secretBulkHandler = async (secretBulkArgs: SecretBulkArgs) => {
 			throw e;
 		}
 	}
-	const inheritBindings = existingBindings.filter((binding) => {
-		return binding.type !== "secret_text" || !content[binding.name];
-	});
+	const inheritBindings = existingBindings
+		.filter(
+			(binding) => binding.type !== "secret_text" || !content[binding.name]
+		)
+		.map((binding) => ({ type: binding.type, name: binding.name }));
```

Comment:
Given that this looks like you are just copying properties like for like over, I think a comment for future readers explaining that you are actually excluding unwanted properties and why would be useful.

Reply:
@petebacondarwin, indeed, this fix addresses the problem I reported in my GitHub issue (https://github.com/cloudflare/workers-sdk/issues/4675). Specifically, the conflict arises from the d1 database binding. The Cloudflare API expects a d1 binding to have a `id` key, but when making a GET request to https://api.cloudflare.com/client/v4/accounts/{account_identifier}/workers/scripts/{script_name}/settings, the response contains a `database_id` key instead of `id` for d1 binding object.